src/test/storage/segment_access_counter_test.cpp

```diff
@@ -0,0 +1,138 @@
+#include <atomic>
+#include <map>
+#include <memory>
+#include <string>
+
+#include "base_test.hpp"
+#include "storage/create_iterable_from_segment.hpp"
+#include "storage/segment_access_counter.hpp"
+#include "storage/table.hpp"
+#include "storage/table_column_definition.hpp"
+#include "storage/value_segment.hpp"
+#include "storage/value_segment/value_segment_iterable.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+template <typename T>
+class SegmentAccessCounterCounterTest : public BaseTest {
+ public:
+  static void SetUpTestCase() {}
+
+  void SetUp() override {}
+
+  void TearDown() override {}
+};
+
+class SegmentAccessCounterTest : public BaseTest {
+ public:
+  static void SetUpTestCase() {}
+
+  void SetUp() override {}
+
+  void TearDown() override {}
+
+ protected:
+  using AccessPattern = SegmentAccessCounter::AccessPattern;
+  using Input = SegmentAccessCounter::Input;
+
+  static AccessPattern _iterator_access_pattern(const std::shared_ptr<const PosList>& positions) {
+    return SegmentAccessCounter::_iterator_access_pattern(positions);
+  }
+};
+
+typedef ::testing::Types<uint64_t, std::atomic_uint64_t> CounterTypes;
+TYPED_TEST_SUITE(SegmentAccessCounterCounterTest, CounterTypes, );  // NOLINT(whitespace/parens)
+
+TYPED_TEST(SegmentAccessCounterCounterTest, ZeroOnConstruction) {
+  SegmentAccessCounter::Counter<TypeParam> counter;
+  EXPECT_EQ(0, counter.sum());
+}
+
+TYPED_TEST(SegmentAccessCounterCounterTest, Sum36) {
+  SegmentAccessCounter::Counter<TypeParam> counter;
+  EXPECT_EQ(0, counter.sum());
+  counter.other = 1;
+  counter.iterator_create = 2;
+  counter.iterator_seq_access = 3;
+  counter.iterator_increasing_access = 4;
+  counter.iterator_random_access = 5;
+  counter.accessor_create = 6;
+  counter.accessor_access = 7;
+  counter.dictionary_access = 8;
+  EXPECT_EQ(36, counter.sum());
```

Comment:
What is this supposed to test? Also, what is sum? Why does it contain the creation of things?